### ISSUE TITLE
Checkbox not compatible with 1.9.6 pygame.draw

### ISSUE THREAD

**Issue by davidzwa** [AustL/PygameWidgets#9]
I understand that the rounded corners of Pygame 2.0.0 are really nice, but it's not released yet. This component is more dependent on it, but I dont mind square checkboxes for now.

I attempted to fix the `selection.py::Checkbox::draw()` function but it didnt show properly.

```
    def draw(self):
        """ Display to surface """
        if not self.hidden:
            for row in range(self.rows):
                colour = self.colour1 if not row % 2 else self.colour2
                if row == 0:
                    draw_rounded_rect(
                        self.win, colour, Rect(self.x, self.y + self.rowHeight * row, self.width, self.rowHeight),
                        border_radius=self.radius #, border_top_right_radius=self.radius
                    )

                elif row == self.rows - 1:
                    draw_rounded_rect(
                        self.win, colour, Rect(self.x, self.y + self.rowHeight * row, self.width, self.rowHeight),
                        border_radius=self.radius #, border_bottom_right_radius=self.radius
                    )

                else:
                    draw_rounded_rect(
                        self.win, colour, Rect(self.x, self.y + self.rowHeight * row, self.width, self.rowHeight)
                    )

                width = 0 if self.selected[row] else self.boxThickness
                # draw_rounded_rect(
                #     self.win, self.boxColour,
                #     self.boxes[row],
                #     border_radius=width
                # )

                self.win.blit(self.texts[row], self.textRects[row])
```

**Comment by davidzwa** [comment]
Had to change so much, I cannot keep track anymore.
```
class Checkbox(WidgetBase):
    def __init__(self, win, x, y, width, height, items, **kwargs):
        """ A collection of buttons

        :param win: Surface on which to draw
        :type win: pygame.Surface
        :param x: X-coordinate of top left
        :type x: int
        :param y: Y-coordinate of top left
        :type y: int
        :param width: Width of button
        :type width: int
        :param height: Height of button
        :type height: int
        :param items: Names of list items
        :type items: tuple of str
        :param kwargs: Optional parameters
        """
        super().__init__(win, x, y, width, height)

        self.items = items
        self.rows = len(items)
        self.rowHeight = self.height // self.rows
        self.selected = [False for _ in range(self.rows)]

        # Border
        self.borderThickness = kwargs.get('borderThickness', 15)
        self.borderColour = kwargs.get('borderColour', (0, 0, 0))
        self.radius = kwargs.get('radius', 0)

        # Checkbox color and size
        self.boxSize = kwargs.get('boxSize', self.height / self.rows // 2)
        self.boxColour = kwargs.get('boxColour', (0, 0, 0))
        self.boxColour2 = kwargs.get('boxColour', (155, 155, 155))

        # Colour background
        self.colour = kwargs.get('colour', (255, 255, 255))
        self.colour2 = kwargs.get('colour', (250, 250, 250))
        self.colour_hover = kwargs.get('colour', (210, 210, 200))

        # Alternating colours: overrides colour
        self.colour1 = kwargs.get('colour1', self.colour)
        self.colour2 = kwargs.get('colour2', self.colour2)

        # Text
        self.textColour = kwargs.get('textColour', (0, 0, 0))
        self.fontSize = kwargs.get('fontSize', 20)
        self.font = kwargs.get('font', pygame.font.SysFont('calibri', self.fontSize))
        self.texts = [self.font.render(self.items[row], True, self.textColour) for row in range(self.rows)]
        self.textRects = self.createTextRects()

        self.clicked = False

        self.boxes = self.createBoxLocations()

    def createTextRects(self):
        textRects = []
        for row in range(self.rows):
            textRects.append(
                self.texts[row].get_rect(
                    center=(
                        self.x + self.boxSize * 2 + (self.width - self.boxSize * 2) // 2,
                        self.y + self.rowHeight * row + self.rowHeight // 2
                    )
                )
            )

        return textRects

    def createBoxLocations(self):
        boxes = []
        for row in range(self.rows):
            boxes.append(pygame.Rect(
                self.x + self.boxSize,
                self.y + self.rowHeight * row + self.boxSize // 2,
                self.boxSize, self.boxSize
            ))
        return boxes

    def listen(self, events):
        """ Wait for inputs

        :param events: Use pygame.event.get()
        :type events: list of pygame.event.Event
        """
        if not self.hidden:
            pressed = pygame.mouse.get_pressed()[0]
            x, y = pygame.mouse.get_pos()

            if self.contains(x, y):
                if pressed:
                    if not self.clicked:
                        self.clicked = True
                        for row in range(self.rows):
                            background_rect = Rect(self.x, self.y + self.rowHeight * row, self.width, self.rowHeight)
                            if background_rect.collidepoint(x, y):
                                self.selected[row] = not self.selected[row]

                elif self.clicked:
                    self.clicked = False
            elif not pressed:
                self.clicked = False

    def draw(self):
        """ Display to surface """
        if not self.hidden:
            for row in range(self.rows):
                idle_color = self.colour1 if not row % 2 else self.colour2
                background_rect = Rect(self.x, self.y + self.rowHeight * row, self.width, self.rowHeight)
                hover = self.check_hover(pygame.mouse.get_pos(), background_rect)
                draw_rounded_rect(
                    self.win, self.colour_hover if hover else idle_color, background_rect,
                    border_radius=self.radius
                )

                draw_rounded_rect(
                    self.win, self.boxColour,
                    self.boxes[row],
                    border_radius=1
                )
                if self.selected[row]:
                    draw_rounded_rect(
                        self.win, self.boxColour2,
                        self.boxes[row],
                        border_radius=5
                    )

                self.win.blit(self.texts[row], self.textRects[row])

    def getSelected(self):
        return [self.items[row] for row in range(self.rows) if self.selected[row]]

    def check_hover(self, mouse_pos: tuple, rect: Rect):
        return rect.contains(Rect(mouse_pos, (1, 1)))
```

**Comment by AustL** [comment]
Fixed and tested working with both pygame 1.9.6 and pygame2.0.0dev
